pkgs/python-by-name/qmcpack/package.nix

```diff
@@ -0,0 +1,98 @@
+{ buildPythonPackage
+, lib
+, fetchFromGitHub
+, boost
+, cmake
+, blas
+, lapack
+, libxml2
+, hdf5-mpi
+, fftw
+, mpi
+, python
+, numpy
+, h5py
+, pandas
+, scipy
+, pyscf
+, mpi4py
+}:
+
+buildPythonPackage rec {
+  pname = "qmcpack";
+  version = "3.17.1";
+
+  src = fetchFromGitHub {
+    owner = "QMCPACK";
+    repo = "qmcpack";
+    rev = "v${version}";
+    sha256 = "sha256-D/wcKULhAsOkyGHN1AlVs3av0yeOvouPMHLcmNmOUo8=";
+  };
+
+  format = "other";
+
+  nativeBuildInputs = [ cmake ];
+
+  buildInputs = [
+    boost
+    blas
+    lapack
+    libxml2
+    hdf5-mpi
+    fftw
+  ];
+
+  cmakeFlags = [
+    "-DQMC_MPI=ON"
+    "-DQMC_OMP=ON"
+    "-DBUILD_AFQMC=ON"
+    "-DCMAKE_C_COMPILER=${lib.getDev mpi}/bin/mpicc"
+    "-DCMAKE_CXX_COMPILER=${lib.getDev mpi}/bin/mpicxx"
+  ];
+
+  propagatedBuildInputs = [
+    mpi
+    numpy
+    h5py
+    pandas
+    scipy
+    pyscf
+    mpi4py
+  ];
+
+  # Install the python utility programs of QMCPACK
+  postInstall =
+    let
+      pythonScripts = [
+        "convert-gamess-ecp-to-qmcpack"
+        "determinants_tools.py"
+      ];
+      perlScripts = [
```

Comment:
The perl scripts are not getting patched properly (`#!/usr/bin/env perl`). May add perl to the inputs?